cashsvr_input_tof.c

```diff
@@ -151,29 +137,86 @@ static int cash_tof_sys_init(bool high_accuracy, int devno, int plen)
 		snprintf(sns_mode_path, plen + LEN_MODE,
 			"%s%d/set_use_case", sysfs_input_str, devno);
 
-		if (chown(sns_mode_path, uid, gid) == -1) {
+		rc = cash_set_permissions(sns_mode_path, "system", "input");
+		if (rc == -1) {
 			ALOGD("Cannot chown %s/enable_ps_sensor", sns_mode_path);
-			return 1;
-		}
-
-		fd = open(sns_mode_path, O_WRONLY);
-		if (fd < 0) {
-			ALOGD("Cannot open %s", sns_mode_path);
-			return 1;
+			free(cash_tof_enable_path);
+			free(sns_mode_path);
+			return rc;
 		}
 
-		rc = write(fd, VL53L0_HIGH_ACCURACY, 1);
+		rc = cash_set_parameter(sns_mode_path, VL53L0_HIGH_ACCURACY, 1);
```

Comment:
`sns_mode_path` Not used after this line. Consider freeing it.

Reply:
Consider `sizeof(VL53L0_HIGH_ACCURACY) - 1` for the size - prevents undefined behaviour when changing the define in the future.